src/metabase/server/middleware/session.clj

```diff
@@ -89,14 +90,14 @@
                           {:max-age (* 60 (config/config-int :max-session-age))})
                         ;; If the authentication request request was made over HTTPS (hopefully always except for
                         ;; local dev instances) add `Secure` attribute so the cookie is only sent over HTTPS.
-                        (when (request.u/https? request)
-                          {:secure true})
-                        (when (= config/mb-session-cookie-samesite :none)
-                          (log/warn
-                           (str (deferred-trs "Session cookie's SameSite is configured to \"None\", but site is")
-                                (deferred-trs "served over an insecure connection. Some browsers will reject ")
-                                (deferred-trs "cookies under these conditions. ")
-                                (deferred-trs "https://www.chromestatus.com/feature/5633521622188032")))))]
+                        (when is-https
+                          {:secure true}))]
+    (when (and (= config/mb-session-cookie-samesite :none) (not is-https))
+      (log/warn
+       (str (deferred-trs "Session cookie's SameSite is configured to \"None\", but site is")
+            (deferred-trs "served over an insecure connection. Some browsers will reject ")
+            (deferred-trs "cookies under these conditions. ")
+            (deferred-trs "https://www.chromestatus.com/feature/5633521622188032"))))
```

Comment:
I know you didn't write this stuff and it was like this before your PR, but I noticed it, so we should just fix it while we're at it.

1.  If we're joining a bunch of strings together with `str` to build a long message we need to put whitespace between the parts. This is going to read `but siteis served` because there's no space after `is`
2. `https://www.chromestatus.com/feature/5633521622188032` should **not** be marked for i18n
3. We shouldn't leave whitespace in the i18n strings, because then we'd have to hope the translators leave the whitespace in there in the translated string. I seriously doubt that is going to happen. So it's better just to do the whitespace separately outside of the i18n
4. It's crazy to ask people to translate in fragments of a sentence. How is one supposed to translate `cookies under these conditions.` ? If we're going to break things up at the very least we should try to do it by sentence

```suggestion
       (str (deferred-trs "Session cookie's SameSite is configured to \"None\", but site is served over an insecure connection.")
            " "
            (deferred-trs "Some browsers will reject cookies under these conditions.")
            " "
            "https://www.chromestatus.com/feature/5633521622188032")))
```

Reply:
Shouldn't we combine the two `deferred-trs` so translators have more context and freedom?

Reply:
The tool we're using doesn't support multi-line i18n forms. This was split up because otherwise the line would be crazy long